test/integration/auth/podsecurity_test.go

```diff
@@ -84,3 +136,113 @@ func startPodSecurityServer(t *testing.T) *kubeapiservertesting.TestServer {
+	}
+	insecureListener, err := net.Listen("tcp", "127.0.0.1:")
+	if err != nil {
+		return "", err
+	}
+	cert, err := dynamiccertificates.NewStaticCertKeyContent("localhost", utiltest.LocalhostCert, utiltest.LocalhostKey)
+	if err != nil {
+		return "", err
+	}
+	defaultConfig, err := podsecurityconfigloader.LoadFromData(nil) // load the default
+	if err != nil {
+		return "", err
+	}
+
+	c := podsecurityserver.Config{
+		SecureServing: &apiserver.SecureServingInfo{
+			Listener: secureListener,
+			Cert:     cert,
+		},
+		InsecureServing: &apiserver.DeprecatedInsecureServingInfo{
+			Listener: insecureListener,
+		},
+		KubeConfig:        testServer.ClientConfig,
+		PodSecurityConfig: defaultConfig,
+	}
+
+	t.Logf("Starting webhook server...")
+	webhookServer, err := podsecurityserver.Setup(&c)
+	if err != nil {
+		return "", err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go webhookServer.Start(ctx)
+	t.Cleanup(cancel)
+
+	// Wait for server to be ready
+	t.Logf("Waiting for webhook server /readyz to be ok...")
+	readyz := (&url.URL{
+		Scheme: "http",
+		Host:   c.InsecureServing.Listener.Addr().String(),
+		Path:   "/readyz",
+	}).String()
+	if err := wait.PollImmediate(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+		resp, err := http.Get(readyz)
+		if err != nil {
+			return false, err
+		}
+		return resp.StatusCode == 200, nil
+	}); err != nil {
+		return "", err
+	}
+
+	return c.SecureServing.Listener.Addr().String(), nil
+}
+
+func installWebhook(clientConfig *rest.Config, addr string) error {
+	client, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return fmt.Errorf("error creating client: %w", err)
+	}
+
+	fail := admissionregistrationv1.Fail
+	equivalent := admissionregistrationv1.Equivalent
+	none := admissionregistrationv1.SideEffectClassNone
+	endpoint := (&url.URL{
+		Scheme: "https",
+		Host:   addr,
+	}).String()
+
+	// Installing Admission webhook to API server
+	_, err = client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Create(context.TODO(), &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{Name: "podsecurity-webhook.integration.test"},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			{
+				Name: "podsecurity-webhook.integration.test",
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					URL:      &endpoint,
+					CABundle: utiltest.LocalhostCert,
+				},
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
+					Rule: admissionregistrationv1.Rule{APIGroups: []string{"*"}, APIVersions: []string{"v1"}, Resources: []string{
+						"namespaces",
+						"pods",
+						"replicationcontrollers",
+						"podtemplates",
+						"replicasets",
+						"deployments",
+						"statefulsets",
+						"daemonsets",
+						"jobs",
+						"cronjobs",
+					}},
```

Comment:
people will copy this, so express the actual groups/version/subresources:

```
Rules: []admissionregistrationv1.RuleWithOperations{
    {
        Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
        Rule: admissionregistrationv1.Rule{
            APIGroups: []string{""},
            APIVersions: []string{"v1"},
            Resources: []string{"namespaces","pods","pods/ephemeralcontainers","replicationcontrollers","podtemplates"},
        },
    },
    {
        Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
        Rule: admissionregistrationv1.Rule{
            APIGroups: []string{"apps"},
            APIVersions: []string{"v1"},
            Resources: []string{"replicasets","deployments","statefulsets","daemonsets"},
        },
    },
    {
        Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
        Rule: admissionregistrationv1.Rule{
            APIGroups: []string{"batch"},
            APIVersions: []string{"v1"},
            Resources: []string{"cronjobs","jobs"},
        },
    },
}
```

Reply:
Thanks for writing it out for my lazy self!